spec/policies/repository_policy_spec.rb

```diff
@@ -0,0 +1,365 @@
+            end
+          end
+
+          context 'as a user role with access' do
+            context 'by ownership' do
+              let(:repository_by_user) do
+                create :repository, owner: user
+              end
+              subject { RepositoryPolicy.new(user, repository_by_user) }
+
+              it 'allows to update the repository' do
+                expect(subject.update?).to be(true)
+              end
+            end
+
+            context 'by repository membership' do
+              subject { RepositoryPolicy.new(user, repository) }
+
+              %w(write read).each do |role|
+                it "with role #{role} does not allow to update the "\
+                  'repository' do
+                  repository.add_member(user, role)
+                  expect(subject.update?).to be(false)
+                end
+              end
+
+              it 'with role admin allows to update the repository' do
+                repository.add_member(user, 'admin')
+                expect(subject.update?).to be(true)
+              end
+            end
+
+            context 'by organization membership' do
+              let(:repository_by_organization) do
+                create :repository, owner: organization
+              end
+              subject do
+                RepositoryPolicy.new(user, repository_by_organization)
+              end
+
+              %w(write read).each do |role|
+                it "with role #{role} does not allow to update the "\
+                  'repository' do
+                  organization.add_member(user, role)
+                  expect(subject.update?).to be(false)
+                end
+              end
+
+              it 'with role admin allows to update the repository' do
+                organization.add_member(user, 'admin')
+                expect(subject.update?).to be(true)
+              end
+            end
+          end
+        end
+      end
+    end
+  end
+
+  context 'index?' do
+    [true, false].each do |public|
+      context "with public access #{public}" do
+        let(:repository) { create :repository, public_access: public }
+
+        context 'not signed in' do
+          subject { RepositoryPolicy.new(nil, repository) }
+
+          it 'allows to list the repository' do
+            expect(subject.index?).to be(true)
+          end
+        end
+
+        context 'signed in' do
+          context 'as user role' do
+            subject { RepositoryPolicy.new(user, repository) }
+
+            it 'allows to list the repository' do
+              expect(subject.index?).to be(true)
+            end
+          end
+
+          context 'as admin role' do
+            subject { RepositoryPolicy.new(admin, repository) }
+
+            it 'allows to list the repository' do
+              expect(subject.index?).to be(true)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  context 'destroy?' do
+    [true, false].each do |public|
```

Comment:
Please name this variable `public_access` instead of `public`.

`public` is a keyword in Ruby. I wonder why line 283 is not a syntax error.